frontend/src/components/TableRatedTaskRow.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+import { Task } from '../redux/roadmaps/types';
+import css from './TableRatedTaskRow.module.scss';
+import { paths } from '../routers/paths';
+import {
+  calcTaskValueSum,
+  calcAverageTaskValue,
+  calcTaskWorkSum,
+  calcAverageTaskWorkSum,
+} from '../utils/TaskUtils';
+
+const classes = classNames.bind(css);
+
+interface TableTaskRowRatedProps {
+  task: Task;
+}
+
+export const TableRatedTaskRow: React.FC<TableTaskRowRatedProps> = ({
+  task,
+}) => {
+  const [hovered, setHovered] = useState(false);
+  return (
+    <tr
+      className={classNames(
+        css.tableRow,
+        hovered ? css.hoverRow : '',
+        task.completed ? css.completedRow : '',
+      )}
+    >
+      <td className={`styledTd ${classes(css.ratedTitle)}`}>
+        {task.completed && <DoneAllIcon className={classes(css.doneIcon)} />}
+        {task.name}
+      </td>
+      <td className={`styledTd ${classes(css.ratedTd)}`}>
+        {calcAverageTaskValue(task) || 0}
```

Comment:
Calculating all the necessary averages and sums could be done in one pass over the task ratings

The utils has the basics in https://github.com/Vincit/VISDOM-Roadmapper/blob/20401ca409e6727156efe86307888cad7180f2a3/frontend/src/utils/TaskUtils.ts#L40-L43

Then the sums and averages is simple to get for any dimension

```js
const { sum, count } = ratings.get(dimension) ?? { sum: 0, count: 0 };
const avg = sum / count;
```